### DISCUSSION TITLE
How RTK Query get cookies in ServerSideProps

### DISCUSSION THREAD

**Discussion by wh5938316** [reduxjs/redux-toolkit#2699] [Q&A]
RTK query cannot get cookies in ServerSideProps。
And I don't know how to pass cookies temporarily or automatically。

**Comment by phryneas** [comment]
RTK Query has no code at all to "get cookies" in any environment. That's not it's job. 
It's very possible that you have written code that would do that, but that doesn't work. Maybe. I don't know. Because I don't see your screen. 

So right now I have no idea what you are doing and what you are missing. You are just putting two sentences without any code examples or explanation here.

Not sure how to help you there, sorry 🤷

**Reply by wh5938316** [reply]
This function in `Next.js` page will return `401 Error`
Because i use `Session` to authenticate user, and it stored in `Cookies`
The Browser will set `Cookies` automatically, but node will not
I don't know if I used `RTK Query` the wrong way in `SSR`

![8051663221909_ pic](https://user-images.githubusercontent.com/16160933/190326791-2545001f-bc59-4596-8ac7-495914308ebc.jpg)

**Reply by phryneas** [reply]
This still has nothing to do with RTK Query at this point.
Yes, if you call `fetch` on node, it will not automatically copy the client's cookies as cookies into that request. `fetch` doesn't even know that there is a client request going on, or if there are cookies involved in any way. You have to write code to extract those cookies from the current request and then pass those cookies into requests that are being made.

As I see no code attempting that whatsoever anywhere: what did you try so far?

**Reply by wh5938316** [reply]
https://github.com/reduxjs/redux-toolkit/issues/2228#issuecomment-1095409011
When I browse issues, I find this answer, but I don't know how to 
>  inject the current `request` into the store thunk's `extra`

I intuitively think this is a good way, but I just started using redux, and I donot know how to achieve it

**Reply by wh5938316** [reply]
I pass `cookies` as query data
<img width="453" alt="image" src="https://user-images.githubusercontent.com/16160933/190335656-f82f3412-129b-4db7-acfb-275936a585d0.png">
And set headers in `query` function
<img width="563" alt="image" src="https://user-images.githubusercontent.com/16160933/190336202-86bcaf60-9fb4-43d0-b242-c15987435c48.png">
It is ugly and it will request in `clientSide` repeatly, because query data is change(no explicit cookies in `clientSide`)
<img width="352" alt="image" src="https://user-images.githubusercontent.com/16160933/190336891-a75d9db7-973e-4327-b4de-dba529bc06d8.png">
